src/aggregate/functions/date.c

```diff
@@ -240,7 +241,7 @@ static int func_month(ExprEval *ctx, RSValue *result, RSValue **argv, size_t arg
   tmm.tm_sec = 0;
   tmm.tm_hour = 0;
   tmm.tm_min = 0;
-  tmm.tm_mday = 1;
+  tmm.tm_yday -= tmm.tm_mday - 1; // set to first day of month
```

Comment:
Is there a more elegant way of fixing this function so that it doesn't expose implementation details to its callsite? Why should the caller of the function be required to set the day of the year to the day of the month?

Reply:
This is OK with our static custom implementation IMO. Partially editing the `tm` struct is not a good practice if the struct has long-term usage because there are many relations between different values, especially for day values.
Changing any of them without changing the others will set the struct to an ambiguous date.
I can add a comment to the `fast_timegm` about what values it uses when calculating the epoch time.